### PR TITLE
remove hardcoded timestep

### PR DESCRIPTION
**Pull request by qyan41**
remove hardcoded timestep in plotting.

### CODE REVIEW

File: gasaerexch/plot_gasaerexch_timestep.py [hunk 1]
```diff
@@ -22,7 +22,36 @@ def __init__(self):
                                                   data.output.qso4_gas_rel, \
                                                   data.output.qsoa_gas_rel
         self.pmid, self.temp, self.dtsubstep = data.input.pmid, data.input.temp,  data.input.dtsubstep
-        
+ 
+        dtrange = set()
+        for time in data.input.dtsubstep:
+            if time not in dtrange:
```
Comment (jeff-cohere) [inline gasaerexch/plot_gasaerexch_timestep.py]: A `set` is an ordered set of unique elements, so you don't need to check whether something is in it before calling `add`. For example, if I have a set `A = {1, 2, 3}`, and I try to add `3` to it, nothing happens. The set maintains the uniqueness of all the elements.

Comment (jeff-cohere) [reply gasaerexch/plot_gasaerexch_timestep.py]: (That `{1, 2, 3}` syntax is the mathematical notation for a set, not Python. 🙂 )

File: gasaerexch/plot_gasaerexch_timestep_update.py [hunk 2]
```diff
@@ -23,7 +23,36 @@ def __init__(self):
                                                   data.output.qso4_gas_rel, \
                                                   data.output.qsoa_gas_rel
         self.pmid, self.temp, self.dtsubstep = data.input.pmid, data.input.temp,  data.input.dtsubstep
-        
+
+        dtrange = set()
+        for time in data.input.dtsubstep:
+            if time not in dtrange:
```
Comment (jeff-cohere) [inline gasaerexch/plot_gasaerexch_timestep_update.py]: As above (and with the other sets), you don't have to perform this check.

**Comment by jeff-cohere** [comment]
Looks good. I just left a couple of comments about sets: you don't need to check for uniqueness yourself--the set does that for you.

File: gasaerexch/plot_gasaerexch_timestep.py (hunk 1 above)
Comment (qyan41) [reply gasaerexch/plot_gasaerexch_timestep.py]: > Looks good. I just left a couple of comments about sets: you don't need to check for uniqueness yourself--the set does that for you.

Thanks for reviewing. I will edit these parts of the code.

File: gasaerexch/plot_gasaerexch_timestep_update.py (hunk 2 above)
Comment (jeff-cohere) [reply gasaerexch/plot_gasaerexch_timestep_update.py]: Is this line 29 still here?

Comment (qyan41) [reply gasaerexch/plot_gasaerexch_timestep_update.py]: removed now.

**Review by jeff-cohere — APPROVED** [review]

**Pull request merged**